src/components/Pagination.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+
+interface PaginationProps {
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+}) => {
+  const maxPagesToShow = 6;
```

Comment:
```suggestion
  const MAX_PAGES_SHOW_TO_SHOW = 6;
```

なんで 6 なのかコメントしておくと良さそう